Uhm, there are [more issues](https://buildkite.com/julialang/enzyme-dot-jl/builds/6938#019dfa2f-c4b6-4131-adbc-7d5fda0e90f2/L311)
```
  AssertionError: Function requesting sret type was not an sret
  idx=1
  fn=define internal fastcc void @julia__sort__164809(ptr addrspace(10) nocapture noundef nonnull readonly align 8 dereferenceable(24) "enzyme_inactive" "enzyme_type"="{[-1]:Pointer, [-1,0]:Pointer, [-1,0,0]:Pointer, [-1,0,8]:Pointer, [-1,0,16]:Integer, [-1,0,17]:Integer, [-1,0,18]:Integer, [-1,0,19]:Integer, [-1,8]:Pointer, [-1,8,0]:Integer, [-1,8,1]:Integer, [-1,8,2]:Integer, [-1,8,3]:Integer, [-1,8,4]:Integer, [-1,8,5]:Integer, [-1,8,6]:Integer, [-1,8,7]:Integer, [-1,8,8]:Pointer, [-1,8,8,0]:Pointer, [-1,8,8,8]:Pointer, [-1,8,8,16]:Integer, [-1,8,8,17]:Integer, [-1,8,8,18]:Integer, [-1,8,8,19]:Integer, [-1,16]:Integer, [-1,17]:Integer, [-1,18]:Integer, [-1,19]:Integer, [-1,20]:Integer, [-1,21]:Integer, [-1,22]:Integer, [-1,23]:Integer}" "enzymejl_parmtype"="140533043354640" "enzymejl_parmtype_ref"="2" "enzymejl_parmtype_str"="Vector{Base.Filesystem.DirEntry}" %"v::Array", ptr addrspace(11) nocapture noundef nonnull readonly align 8 dereferenceable(24) "enzyme_inactive" "enzyme_type"="{[-1]:Pointer, [-1,0]:Pointer, [-1,0,0]:Pointer, [-1,0,0,0]:Pointer, [-1,0,0,8]:Pointer, [-1,0,0,16]:Integer, [-1,0,0,17]:Integer, [-1,0,0,18]:Integer, [-1,0,0,19]:Integer, [-1,0,8]:Pointer, [-1,0,8,0]:Integer, [-1,0,8,1]:Integer, [-1,0,8,2]:Integer, [-1,0,8,3]:Integer, [-1,0,8,4]:Integer, [-1,0,8,5]:Integer, [-1,0,8,6]:Integer, [-1,0,8,7]:Integer, [-1,0,8,8]:Pointer, [-1,0,8,8,0]:Pointer, [-1,0,8,8,8]:Pointer, [-1,0,8,8,16]:Integer, [-1,0,8,8,17]:Integer, [-1,0,8,8,18]:Integer, [-1,0,8,8,19]:Integer, [-1,0,16]:Integer, [-1,0,17]:Integer, [-1,0,18]:Integer, [-1,0,19]:Integer, [-1,0,20]:Integer, [-1,0,21]:Integer, [-1,0,22]:Integer, [-1,0,23]:Integer, [-1,8]:Integer, [-1,9]:Integer, [-1,10]:Integer, [-1,11]:Integer, [-1,12]:Integer, [-1,13]:Integer, [-1,14]:Integer, [-1,15]:Integer, [-1,16]:Integer, [-1,17]:Integer, [-1,18]:Integer, [-1,19]:Integer, [-1,20]:Integer, [-1,21]:Integer, [-1,22]:Integer, [-1,23]:Integer}" "enzymejl_parmtype"="140533099042784" "enzymejl_parmtype_ref"="1" "enzymejl_parmtype_str"="@NamedTuple{scratch::Vector{Base.Filesystem.DirEntry}, lo::Int64, hi::Int64}" %"kw::NamedTuple", ptr nocapture readonly "enzyme_inactive" "enzyme_type"="{[-1]:Pointer, [-1,0]:Pointer, [-1,0,0]:Pointer, [-1,0,0,0]:Pointer, [-1,0,0,8]:Pointer, [-1,0,0,16]:Integer, [-1,0,0,17]:Integer, [-1,0,0,18]:Integer, [-1,0,0,19]:Integer, [-1,0,8]:Pointer, [-1,0,8,0]:Integer, [-1,0,8,1]:Integer, [-1,0,8,2]:Integer, [-1,0,8,3]:Integer, [-1,0,8,4]:Integer, [-1,0,8,5]:Integer, [-1,0,8,6]:Integer, [-1,0,8,7]:Integer, [-1,0,8,8]:Pointer, [-1,0,8,8,0]:Pointer, [-1,0,8,8,8]:Pointer, [-1,0,8,8,16]:Integer, [-1,0,8,8,17]:Integer, [-1,0,8,8,18]:Integer, [-1,0,8,8,19]:Integer, [-1,0,16]:Integer, [-1,0,17]:Integer, [-1,0,18]:Integer, [-1,0,19]:Integer, [-1,0,20]:Integer, [-1,0,21]:Integer, [-1,0,22]:Integer, [-1,0,23]:Integer}" "enzymejl_parmtype"="140528492572432" "enzymejl_parmtype_ref"="1" "enzymejl_parmtype_str"="@NamedTuple{var\221\22::Vector{Base.Filesystem.DirEntry}}" "enzymejl_rooted_typ"="140533099042784" %.roots.kw) unnamed_addr #1975 !dbg !193258 {
  top:
    %pgcstack = call ptr @julia.get_pgcstack()
    %0 = load ptr addrspace(10), ptr %.roots.kw, align 8, !tbaa !1734, !alias.scope !1824, !noalias !1825
    %ptls_field = getelementptr inbounds i8, ptr %pgcstack, i64 16
    %ptls_load = load ptr, ptr %ptls_field, align 8, !tbaa !1734
    %1 = getelementptr inbounds i8, ptr %ptls_load, i64 16
    %safepoint = load ptr, ptr %1, align 8, !tbaa !1738
    fence syncscope("singlethread") seq_cst
    call void @julia.safepoint(ptr %safepoint), !dbg !193259
    fence syncscope("singlethread") seq_cst
    %2 = getelementptr inbounds i8, ptr addrspace(11) %"kw::NamedTuple", i64 8, !dbg !193260
    %3 = getelementptr inbounds i8, ptr addrspace(11) %"kw::NamedTuple", i64 16, !dbg !193265
    %.unbox = load i64, ptr addrspace(11) %2, align 8, !dbg !193268, !tbaa !1738, !alias.scope !1770, !noalias !1771
    %4 = add i64 %.unbox, 1, !dbg !193268
    %.unbox1 = load i64, ptr addrspace(11) %3, align 8, !dbg !193271
    %.not = icmp sgt i64 %4, %.unbox1, !dbg !193271
    %.unbox..unbox1 = select i1 %.not, i64 %.unbox, i64 %.unbox1, !dbg !193275
    %.not110 = icmp slt i64 %.unbox..unbox1, %4, !dbg !193282
    br i1 %.not110, label %L195, label %L19.L28_crit_edge, !dbg !193281
  L19.L28_crit_edge:                                ; preds = %top
    %.phi.trans.insert = addrspacecast ptr addrspace(10) %"v::Array" to ptr addrspace(11)
    %memoryref_data.pre = load ptr, ptr addrspace(11) %.phi.trans.insert, align 8, !dbg !193290, !tbaa !1782, !alias.scope !1785, !noalias !1786
    %.phi.trans.insert97 = getelementptr inbounds { ptr, ptr addrspace(10) }, ptr addrspace(11) %.phi.trans.insert, i64 0, i32 1
    %memoryref_mem.pre = load ptr addrspace(10), ptr addrspace(11) %.phi.trans.insert97, align 8, !dbg !193290, !tbaa !1782, !alias.scope !1785, !noalias !1786
    %jl_undefref_exception.pre = load ptr, ptr @jl_undefref_exception, align 8, !dbg !193290, !tbaa !1738, !alias.scope !1770, !noalias !1771
    br label %L28, !dbg !193281
  L28:                                              ; preds = %L179, %L19.L28_crit_edge
    %memoryref_mem45108 = phi ptr addrspace(10) [ %memoryref_mem.pre, %L19.L28_crit_edge ], [ %memoryref_mem45, %L179 ]
    %memoryref_data36105 = phi ptr [ %memoryref_data.pre, %L19.L28_crit_edge ], [ %memoryref_data36, %L179 ]
    %value_phi5 = phi i64 [ %4, %L19.L28_crit_edge ], [ %41, %L179 ]
    %5 = mul i64 %value_phi5, 24, !dbg !193290
    %6 = call ptr addrspace(13) @julia.gc_loaded(ptr addrspace(10) %memoryref_mem45108, ptr %memoryref_data36105), !dbg !193290
    %7 = getelementptr i8, ptr addrspace(13) %6, i64 %5, !dbg !193290
    %memoryref_data7 = getelementptr i8, ptr addrspace(13) %7, i64 -24, !dbg !193290
    %8 = load atomic ptr addrspace(10), ptr addrspace(13) %memoryref_data7 unordered, align 8, !dbg !193290, !tbaa !1997, !alias.scope !1809, !noalias !1810
    %9 = getelementptr i8, ptr addrspace(13) %7, i64 -16, !dbg !193290
    %10 = load atomic ptr addrspace(10), ptr addrspace(13) %9 unordered, align 8, !dbg !193290, !tbaa !1997, !alias.scope !1809, !noalias !1810
    %11 = getelementptr i8, ptr addrspace(13) %7, i64 -8, !dbg !193290
    %.sroa.293.16.copyload = load i64, ptr addrspace(13) %11, align 8, !dbg !193290, !tbaa !1748, !alias.scope !1926, !noalias !193293
    %.not111 = icmp eq ptr addrspace(10) %8, null, !dbg !193290
    br i1 %.not111, label %fail, label %L49.preheader, !dbg !193290
  L49.preheader:                                    ; preds = %L28
    %.unbox9138 = load i64, ptr addrspace(11) %2, align 8, !dbg !193296, !tbaa !1738, !alias.scope !1770, !noalias !1771
    %.not112139 = icmp slt i64 %.unbox9138, %value_phi5, !dbg !193296
    br i1 %.not112139, label %L52.lr.ph, label %L49.L179_crit_edge, !dbg !193298
  L52.lr.ph:                                        ; preds = %L49.preheader
    %12 = addrspacecast ptr addrspace(10) %8 to ptr addrspace(11)
    %13 = call nonnull ptr @julia.pointer_from_objref(ptr addrspace(11) %12) #1978, !dbg !193299
    %string_ptr72 = getelementptr inbounds i8, ptr %13, i64 8
    %14 = addrspacecast ptr addrspace(10) %10 to ptr addrspace(11)
    %15 = call nonnull ptr @julia.pointer_from_objref(ptr addrspace(11) %14) #1978, !dbg !193299
    %string_ptr = getelementptr inbounds i8, ptr %15, i64 8
    br label %L52, !dbg !193298
  L49.L179_crit_edge:                               ; preds = %L158, %L49.preheader
    %memoryref_mem45107.lcssa = phi ptr addrspace(10) [ %memoryref_mem45108, %L49.preheader ], [ %memoryref_mem45.pre128, %L158 ]
    %memoryref_data36104.lcssa = phi ptr [ %memoryref_data36105, %L49.preheader ], [ %memoryref_data36.pre127, %L158 ]
    %value_phi8.lcssa = phi i64 [ %value_phi5, %L49.preheader ], [ %16, %L158 ]
    %.pre = add i64 %value_phi8.lcssa, -1, !dbg !193300
    br label %L179, !dbg !193298
  L52:                                              ; preds = %L158, %L52.lr.ph
    %value_phi8142 = phi i64 [ %value_phi5, %L52.lr.ph ], [ %16, %L158 ]
    %memoryref_data36104141 = phi ptr [ %memoryref_data36105, %L52.lr.ph ], [ %memoryref_data36.pre127, %L158 ]
    %memoryref_mem45107140 = phi ptr addrspace(10) [ %memoryref_mem45108, %L52.lr.ph ], [ %memoryref_mem45.pre128, %L158 ]
    %16 = add nsw i64 %value_phi8142, -1, !dbg !193305
    %17 = mul i64 %value_phi8142, 24, !dbg !193308
    %18 = call ptr addrspace(13) @julia.gc_loaded(ptr addrspace(10) %memoryref_mem45107140, ptr %memoryref_data36104141), !dbg !193308
    %19 = getelementptr i8, ptr addrspace(13) %18, i64 %17, !dbg !193308
    %memoryref_data22 = getelementptr i8, ptr addrspace(13) %19, i64 -48, !dbg !193308
    %20 = load atomic ptr addrspace(10), ptr addrspace(13) %memoryref_data22 unordered, align 8, !dbg !193308, !tbaa !1997, !alias.scope !1809, !noalias !1810
    %21 = getelementptr i8, ptr addrspace(13) %19, i64 -40, !dbg !193308
    %22 = load atomic ptr addrspace(10), ptr addrspace(13) %21 unordered, align 8, !dbg !193308, !tbaa !1997, !alias.scope !1809, !noalias !1810
    %23 = getelementptr i8, ptr addrspace(13) %19, i64 -32, !dbg !193308
    %.sroa.2.16.copyload = load i64, ptr addrspace(13) %23, align 8, !dbg !193308, !tbaa !1748, !alias.scope !1926, !noalias !193293
    %.not113 = icmp eq ptr addrspace(10) %20, null, !dbg !193308
    br i1 %.not113, label %fail24, label %pass25, !dbg !193308
  L75:                                              ; preds = %guard_pass, %pass25
    %sizeof = load i64, ptr addrspace(10) %10, align 8, !dbg !193309, !tbaa !1738, !range !3198, !alias.scope !1770, !noalias !1771
    %sizeof26 = load i64, ptr addrspace(10) %22, align 8, !dbg !193309, !tbaa !1738, !range !3198, !alias.scope !1770, !noalias !1771
    %24 = call i64 @llvm.umin.i64(i64 %sizeof26, i64 %sizeof), !dbg !193320
    %25 = call token (...) @llvm.julia.gc_preserve_begin(ptr addrspace(10) %10, ptr addrspace(10) %22), !dbg !193328
    %26 = addrspacecast ptr addrspace(10) %22 to ptr addrspace(11), !dbg !193329
    %27 = call nonnull ptr @julia.pointer_from_objref(ptr addrspace(11) %26) #1978, !dbg !193329
    %string_ptr29 = getelementptr inbounds i8, ptr %27, i64 8, !dbg !193329
    %28 = call i32 @memcmp(ptr nonnull %string_ptr, ptr nonnull %string_ptr29, i64 %24), !dbg !193332
    call void @llvm.julia.gc_preserve_end(token %25), !dbg !193334
    %29 = icmp sgt i32 %28, -1, !dbg !193335
    br i1 %29, label %L95, label %L141.thread, !dbg !193336
  L95:                                              ; preds = %L75
    %.not116 = icmp eq i32 %28, 0, !dbg !193337
    %.not117 = icmp ult i64 %sizeof, %sizeof26
    %spec.select = select i1 %.not116, i1 %.not117, i1 false, !dbg !193336
    %memoryref_data36.pre129 = load ptr, ptr addrspace(11) %.phi.trans.insert, align 8, !dbg !193339, !tbaa !1782, !alias.scope !1785, !noalias !1786
    %memoryref_mem45.pre130 = load ptr addrspace(10), ptr addrspace(11) %.phi.trans.insert97, align 8, !dbg !193339, !tbaa !1782, !alias.scope !1785, !noalias !1786
    br i1 %spec.select, label %L158, label %L179, !dbg !193319
  L107:                                             ; preds = %guard_pass
    %sizeof67 = load i64, ptr addrspace(10) %8, align 8, !dbg !193309, !tbaa !1738, !range !3198, !alias.scope !1770, !noalias !1771
    %sizeof68 = load i64, ptr addrspace(10) %20, align 8, !dbg !193309, !tbaa !1738, !range !3198, !alias.scope !1770, !noalias !1771
    %30 = call i64 @llvm.umin.i64(i64 %sizeof68, i64 %sizeof67), !dbg !193320
    %31 = call token (...) @llvm.julia.gc_preserve_begin(ptr addrspace(10) nonnull %8, ptr addrspace(10) nonnull %20), !dbg !193328
    %32 = call nonnull ptr @julia.pointer_from_objref(ptr addrspace(11) %44) #1978, !dbg !193329
    %string_ptr73 = getelementptr inbounds i8, ptr %32, i64 8, !dbg !193329
    %33 = call i32 @memcmp(ptr nonnull %string_ptr72, ptr nonnull %string_ptr73, i64 %30), !dbg !193332
    call void @llvm.julia.gc_preserve_end(token %31), !dbg !193334
    %34 = icmp sgt i32 %33, -1, !dbg !193335
    br i1 %34, label %L141, label %L141.thread, !dbg !193336
  L141.thread:                                      ; preds = %L107, %L75
    %memoryref_data36.pre125 = load ptr, ptr addrspace(11) %.phi.trans.insert, align 8, !dbg !193339, !tbaa !1782, !alias.scope !1785, !noalias !1786
    %memoryref_mem45.pre126 = load ptr addrspace(10), ptr addrspace(11) %.phi.trans.insert97, align 8, !dbg !193339, !tbaa !1782, !alias.scope !1785, !noalias !1786
    br label %L158, !dbg !193319
  L141:                                             ; preds = %L107
    %.not119 = icmp eq i32 %33, 0, !dbg !193337
    %.not120 = icmp ult i64 %sizeof67, %sizeof68
    %spec.select122 = select i1 %.not119, i1 %.not120, i1 false, !dbg !193336
    %memoryref_data36.pre = load ptr, ptr addrspace(11) %.phi.trans.insert, align 8, !dbg !193339, !tbaa !1782, !alias.scope !1785, !noalias !1786
    %memoryref_mem45.pre = load ptr addrspace(10), ptr addrspace(11) %.phi.trans.insert97, align 8, !dbg !193339, !tbaa !1782, !alias.scope !1785, !noalias !1786
    br i1 %spec.select122, label %L158, label %L179, !dbg !193319
  L158:                                             ; preds = %L141, %L141.thread, %L95
    %memoryref_mem45.pre128 = phi ptr addrspace(10) [ %memoryref_mem45.pre126, %L141.thread ], [ %memoryref_mem45.pre, %L141 ], [ %memoryref_mem45.pre130, %L95 ]
    %memoryref_data36.pre127 = phi ptr [ %memoryref_data36.pre125, %L141.thread ], [ %memoryref_data36.pre, %L141 ], [ %memoryref_data36.pre129, %L95 ]
    %memoryref_byteoffset56 = mul i64 %16, 24, !dbg !193341
    %35 = call ptr addrspace(13) @julia.gc_loaded(ptr addrspace(10) %memoryref_mem45.pre128, ptr %memoryref_data36.pre127), !dbg !193341
    %memoryref_data61 = getelementptr inbounds i8, ptr addrspace(13) %35, i64 %memoryref_byteoffset56, !dbg !193341
    store atomic ptr addrspace(10) %20, ptr addrspace(13) %memoryref_data61 unordered, align 8, !dbg !193341, !tbaa !1997, !alias.scope !1809, !noalias !193344
    %36 = getelementptr inbounds i8, ptr addrspace(13) %memoryref_data61, i64 8, !dbg !193341
    store atomic ptr addrspace(10) %22, ptr addrspace(13) %36 unordered, align 8, !dbg !193341, !tbaa !1997, !alias.scope !1809, !noalias !193344
    %37 = getelementptr inbounds i8, ptr addrspace(13) %memoryref_data61, i64 16, !dbg !193341
    store i64 %.sroa.2.16.copyload, ptr addrspace(13) %37, align 8, !dbg !193341, !tbaa !1748, !alias.scope !2173, !noalias !193293
    call void (ptr addrspace(10), ...) @julia.write_barrier(ptr addrspace(10) nonnull %memoryref_mem45.pre128, ptr addrspace(10) nonnull %20, ptr addrspace(10) %22), !dbg !193341
    %.unbox9 = load i64, ptr addrspace(11) %2, align 8, !dbg !193296, !tbaa !1738, !alias.scope !1770, !noalias !1771
    %.not112 = icmp slt i64 %.unbox9, %16, !dbg !193296
    br i1 %.not112, label %L52, label %L49.L179_crit_edge, !dbg !193298
  L179:                                             ; preds = %L141, %L95, %L49.L179_crit_edge
    %memoryref_offset38.pre-phi = phi i64 [ %.pre, %L49.L179_crit_edge ], [ %16, %L141 ], [ %16, %L95 ], !dbg !193300
    %memoryref_mem45 = phi ptr addrspace(10) [ %memoryref_mem45107.lcssa, %L49.L179_crit_edge ], [ %memoryref_mem45.pre, %L141 ], [ %memoryref_mem45.pre130, %L95 ], !dbg !193339
    %memoryref_data36 = phi ptr [ %memoryref_data36104.lcssa, %L49.L179_crit_edge ], [ %memoryref_data36.pre, %L141 ], [ %memoryref_data36.pre129, %L95 ], !dbg !193339
    %memoryref_byteoffset39 = mul i64 %memoryref_offset38.pre-phi, 24, !dbg !193300
    %38 = call ptr addrspace(13) @julia.gc_loaded(ptr addrspace(10) %memoryref_mem45, ptr %memoryref_data36), !dbg !193300
    %memoryref_data44 = getelementptr inbounds i8, ptr addrspace(13) %38, i64 %memoryref_byteoffset39, !dbg !193300
    store atomic ptr addrspace(10) %8, ptr addrspace(13) %memoryref_data44 unordered, align 8, !dbg !193300, !tbaa !1997, !alias.scope !1809, !noalias !193344
    %39 = getelementptr inbounds i8, ptr addrspace(13) %memoryref_data44, i64 8, !dbg !193300
    store atomic ptr addrspace(10) %10, ptr addrspace(13) %39 unordered, align 8, !dbg !193300, !tbaa !1997, !alias.scope !1809, !noalias !193344
    %40 = getelementptr inbounds i8, ptr addrspace(13) %memoryref_data44, i64 16, !dbg !193300
    store i64 %.sroa.293.16.copyload, ptr addrspace(13) %40, align 8, !dbg !193300, !tbaa !1748, !alias.scope !2173, !noalias !193293
    call void (ptr addrspace(10), ...) @julia.write_barrier(ptr addrspace(10) nonnull %memoryref_mem45, ptr addrspace(10) nonnull %8, ptr addrspace(10) %10), !dbg !193300
    %.not121 = icmp eq i64 %value_phi5, %.unbox..unbox1, !dbg !193345
    %41 = add i64 %value_phi5, 1, !dbg !193347
    br i1 %.not121, label %L195, label %L28, !dbg !193348
  L195:                                             ; preds = %L179, %top
    ret void, !dbg !193348
  fail:                                             ; preds = %L28
    %42 = addrspacecast ptr %jl_undefref_exception.pre to ptr addrspace(12), !dbg !193290
    call void @ijl_throw(ptr addrspace(12) %42), !dbg !193290
    unreachable, !dbg !193290
  fail24:                                           ; preds = %L52
    %43 = addrspacecast ptr %jl_undefref_exception.pre to ptr addrspace(12), !dbg !193308
    call void @ijl_throw(ptr addrspace(12) %43), !dbg !193308
    unreachable, !dbg !193308
  pass25:                                           ; preds = %L52
    %44 = addrspacecast ptr addrspace(10) %20 to ptr addrspace(11), !dbg !193349
    %.not114 = icmp eq ptr addrspace(11) %12, %44, !dbg !193349
    br i1 %.not114, label %L75, label %guard_pass, !dbg !193349
  guard_pass:                                       ; preds = %pass25
    %45 = call i32 @jl_egal__unboxed(ptr addrspace(11) %12, ptr addrspace(11) %44, i64 160), !dbg !193349
    %46 = and i32 %45, 1, !dbg !193349
    %.not123 = icmp eq i32 %46, 0, !dbg !193349
    br i1 %.not123, label %L107, label %L75, !dbg !193315
  }
   enzymejl_parmtype=Vector{Base.Filesystem.DirEntry} enzymejl_parmtype_ref=MUT_REF
  Stacktrace:
    [1] sret_ty(fn::LLVM.Function, idx::Int64)
      @ Enzyme /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/utils.jl:600
    [2] propagate_returned!(mod::LLVM.Module)
      @ Enzyme.Compiler /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/llvm/transforms.jl:2055
    [3] removeDeadArgs!(mod::LLVM.Module, tm::LLVM.TargetMachine, post_gc_fixup::Bool)
      @ Enzyme.Compiler /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/llvm/transforms.jl:2761
    [4] optimize!(mod::LLVM.Module, tm::LLVM.TargetMachine)
      @ Enzyme.Compiler /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/compiler/optimize.jl:200
    [5] nested_codegen!(mode::Enzyme.API.CDerivativeMode, mod::LLVM.Module, funcspec::Core.MethodInstance, world::UInt64)
      @ Enzyme.Compiler /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/compiler.jl:1372
    [6] enzyme_custom_common_rev(forward::Bool, B::LLVM.IRBuilder, orig::LLVM.CallInst, gutils::Enzyme.Compiler.GradientUtils, normalR::Ptr{Ptr{LLVM.API.LLVMOpaqueValue}}, shadowR::Ptr{Ptr{LLVM.API.LLVMOpaqueValue}}, tape::Nothing)
      @ Enzyme.Compiler /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/rules/customrules.jl:1641
    [7] enzyme_custom_augfwd
      @ /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/rules/customrules.jl:2435 [inlined]
    [8] enzyme_custom_augfwd_cfunc(B::Ptr{LLVM.API.LLVMOpaqueBuilder}, OrigCI::Ptr{LLVM.API.LLVMOpaqueValue}, gutils::Ptr{Nothing}, normalR::Ptr{Ptr{LLVM.API.LLVMOpaqueValue}}, shadowR::Ptr{Ptr{LLVM.API.LLVMOpaqueValue}}, tapeR::Ptr{Ptr{LLVM.API.LLVMOpaqueValue}})
      @ Enzyme.Compiler /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/rules/llvmrules.jl:18
    [9] EnzymeCreatePrimalAndGradient(logic::Enzyme.Logic, todiff::LLVM.Function, retType::Enzyme.API.CDIFFE_TYPE, constant_args::Vector{Enzyme.API.CDIFFE_TYPE}, TA::Enzyme.TypeAnalysis, returnValue::Bool, dretUsed::Bool, mode::Enzyme.API.CDerivativeMode, runtimeActivity::Bool, strongZero::Bool, width::Int64, additionalArg::Ptr{Nothing}, forceAnonymousTape::Bool, typeInfo::Enzyme.FnTypeInfo, uncacheable_args::Vector{Bool}, augmented::Ptr{Nothing}, atomicAdd::Bool)
      @ Enzyme.API /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/api.jl:270
   [10] macro expansion
      @ /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/compiler.jl:2783 [inlined]
   [11] macro expansion
      @ ~/.cache/julia-buildkite-plugin/depots/260b8bb4-e043-429b-b2f1-bbf9045c057a/packages/LLVM/Bd7kA/src/base.jl:113 [inlined]
   [12] enzyme!(job::GPUCompiler.CompilerJob{Enzyme.Compiler.EnzymeTarget{GPUCompiler.NativeCompilerTarget}, Enzyme.Compiler.EnzymeCompilerParams{Enzyme.Compiler.PrimalCompilerParams}}, interp::Enzyme.Compiler.Interpreter.EnzymeInterpreter{Nothing}, mod::LLVM.Module, primalf::LLVM.Function, TT::Type, mode::Enzyme.API.CDerivativeMode, width::Int64, parallel::Bool, actualRetType::Type, wrap::Bool, modifiedBetween::NTuple{N, Bool} where N, returnPrimal::Bool, expectedTapeType::Type, loweredArgs::Set{Int64}, boxedArgs::Set{Int64}, removedRoots::Set{Int64})
      @ Enzyme.Compiler /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/compiler.jl:2656
   [13] compile_unhooked(output::Symbol, job::GPUCompiler.CompilerJob{Enzyme.Compiler.EnzymeTarget{GPUCompiler.NativeCompilerTarget}, Enzyme.Compiler.EnzymeCompilerParams{Enzyme.Compiler.PrimalCompilerParams}})
      @ Enzyme.Compiler /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/compiler.jl:5863
   [14] #compile#94
      @ ~/.cache/julia-buildkite-plugin/depots/260b8bb4-e043-429b-b2f1-bbf9045c057a/packages/GPUCompiler/lHkad/src/driver.jl:67 [inlined]
   [15] compile
      @ ~/.cache/julia-buildkite-plugin/depots/260b8bb4-e043-429b-b2f1-bbf9045c057a/packages/GPUCompiler/lHkad/src/driver.jl:55 [inlined]
   [16] _thunk(job::GPUCompiler.CompilerJob{Enzyme.Compiler.EnzymeTarget{GPUCompiler.NativeCompilerTarget}, Enzyme.Compiler.EnzymeCompilerParams{Enzyme.Compiler.PrimalCompilerParams}}, postopt::Bool)
      @ Enzyme.Compiler /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/compiler.jl:6776
   [17] _thunk
      @ /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/compiler.jl:6774 [inlined]
   [18] cached_compilation
      @ /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/compiler.jl:6832 [inlined]
   [19] thunkbase(mi::Core.MethodInstance, World::UInt64, FA::Type{<:Annotation}, A::Type{<:Annotation}, TT::Type, Mode::Enzyme.API.CDerivativeMode, width::Int64, ModifiedBetween::NTuple{N, Bool} where N, ReturnPrimal::Bool, ShadowInit::Bool, ABI::Type, ErrIfFuncWritten::Bool, RuntimeActivity::Bool, StrongZero::Bool, edges::Vector{Any})
      @ Enzyme.Compiler /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/compiler.jl:6948
   [20] thunk_generator(world::UInt64, source::Union{LineNumberNode, Method}, FA::Type, A::Type, TT::Type, Mode::Enzyme.API.CDerivativeMode, Width::Int64, ModifiedBetween::NTuple{N, Bool} where N, ReturnPrimal::Bool, ShadowInit::Bool, ABI::Type, ErrIfFuncWritten::Bool, RuntimeActivity::Bool, StrongZero::Bool, self::Any, fakeworld::Any, fa::Type, a::Type, tt::Type, mode::Type, width::Type, modifiedbetween::Type, returnprimal::Type, shadowinit::Type, abi::Type, erriffuncwritten::Type, runtimeactivity::Type, strongzero::Type)
      @ Enzyme.Compiler /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/compiler.jl:7092
   [21] autodiff
      @ /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/Enzyme.jl:509 [inlined]
   [22] autodiff
      @ /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/Enzyme.jl:569 [inlined]
   [23] autodiff(mode::ReverseMode{false, false, false, FFIABI, false, false}, f::typeof(square!), args::Duplicated{CuArray{Float32, 1, CUDACore.DeviceMemory}})
      @ Enzyme /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/src/Enzyme.jl:541
   [24] top-level scope
      @ /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/test/cuda.jl:179
   [25] macro expansion
      @ ~/.cache/julia-buildkite-plugin/julia_installs/bin/linux/x64/1.12/julia-1.12-latest-linux-x86_64/share/julia/stdlib/v1.12/Test/src/Test.jl:1777 [inlined]
   [26] macro expansion
      @ /var/lib/buildkite-agent/builds/gpuci-6/julialang/enzyme-dot-jl/test/cuda.jl:183 [inlined]
   [27] include(mod::Module, _path::String)
      @ Base ./Base.jl:306
   [28] exec_options(opts::Base.JLOptions)
      @ Base ./client.jl:317
   [29] _start()
      @ Base ./client.jl:550
Test Summary:  | Error  Total     Time
Reverse Kernel |     1      1  6m21.4s
```

--- a/.buildkite/pipeline.yml
+++ b/.buildkite/pipeline.yml
@@ -20,7 +20,7 @@ steps:
           using Pkg
           pkgs = [PackageSpec(; path) for path in (".", "lib/EnzymeCore", "lib/EnzymeTestUtils")]
           Pkg.develop(pkgs)
-          Pkg.add(PackageSpec(; name="CUDA", version="5"))' || exit 3
+          Pkg.add(PackageSpec(; name="CUDA", version="6"))' || exit 3
 
       echo "+++ Run tests"
       julia --color=yes test/cuda.jl

